### ISSUE TITLE
PANIC in query-engine/core/src/interpreter/query_interpreters/nested_read.rs:79:561

### ISSUE THREAD

**Issue by Hazeman28** [prisma/prisma#7447]
Hi Prisma Team! My Prisma Client just crashed. This is the report:
## Versions

| Name            | Version            |
|-----------------|--------------------|
| Node            | v14.15.4           | 
| OS              | darwin             |
| Prisma Client   | 2.23.0             |
| Query Engine    | query-engine adf5e8cba3daf12d456d911d72b6e9418681b28b|
| Database        | undefined|

## Query
```
query {
  findManyTournamentPrize(where: {
    tournamentId: 5
  }) {
    id
    createdAt
    customTypeId
    currencyId
    type
    placementType
    quantity
    placement
    state
    tournamentId
    currency {
      id
      createdAt
      name
      code
      icon
    }
    customType {
      id
      createdAt
      name
    }
    winners {
      tournamentId
      name
      players
      contact
      isManager
      createdAt
      schedule
      extra
      discord
      pending
      state
      id
      fixedOrder
      ownerId
      manager
      custom
      teamId
    }
  }
}
```

## Logs
```
 
  prisma:engine { flags: [ '--enable-raw-queries', '--port', '53312' ] }  
  prisma:engine stdout  Starting a postgresql pool with 5 connections.  
  prisma:engine stdout  Started http server on http://127.0.0.1:53312  
  prisma:engine Search for Query Engine in /Users/hazeman/Development/uniit/ygames/node_modules/.prisma/client  
  plusX Execution permissions of /Users/hazeman/Development/uniit/ygames/node_modules/.prisma/client/query-engine-darwin are fine  
  prisma:engine Client Version: 2.23.0  
  prisma:engine Engine Version: query-engine adf5e8cba3daf12d456d911d72b6e9418681b28b  
  prisma:engine Active provider: postgresql  
  prisma:engine stdout  PANIC in query-engine/core/src/interpreter/query_interpreters/nested_read.rs:79:56
1  +36s
  prisma:engine {
  prisma:engine   error: SocketError: other side closed
  prisma:engine       at Socket.onSocketEnd (/Users/hazeman/Development/uniit/ygames/node_modules/@prisma/client/runtime/index.js:25530:24)
  prisma:engine       at Socket.emit (events.js:327:22)
  prisma:engine       at endReadableNT (internal/streams/readable.js:1327:12)
  prisma:engine       at processTicksAndRejections (internal/process/task_queues.js:80:21) {
  prisma:engine     code: 'UND_ERR_SOCKET'
  prisma:engine   }
  prisma:engine }  
  prisma:engine {
  prisma:engine   error: Error: connect ECONNREFUSED 127.0.0.1:53312
  prisma:engine       at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:16) {
  prisma:engine     errno: -61,
  prisma:engine     code: 'ECONNREFUSED',
  prisma:engine     syscall: 'connect',
  prisma:engine     address: '127.0.0.1',
  prisma:engine     port: 53312
  prisma:engine   }
  prisma:engine }  
  prisma:engine { cwd: '/Users/hazeman/Development/uniit/ygames/prisma' }  
  prisma:engine Search for Query Engine in /Users/hazeman/Development/uniit/ygames/node_modules/.prisma/client  
  plusX Execution permissions of /Users/hazeman/Development/uniit/ygames/node_modules/.prisma/client/query-engine-darwin are fine  +36s
  prisma:engine { flags: [ '--enable-raw-queries', '--port', '53426' ] }  
  prisma:engine {
  prisma:engine   error: Error: connect ECONNREFUSED 127.0.0.1:53312
  prisma:engine       at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:16) {
  prisma:engine     errno: -61,
  prisma:engine     code: 'ECONNREFUSED',
  prisma:engine     syscall: 'connect',
  prisma:engine     address: '127.0.0.1',
  prisma:engine     port: 53312
  prisma:engine   }
  prisma:engine }  
  prisma:engine stdout  Starting a postgresql pool with 5 connections.  
  prisma:engine stdout  Started http server on http://127.0.0.1:53426  
  prisma:engine Search for Query Engine in /Users/hazeman/Development/uniit/ygames/node_modules/.prisma/client  
  plusX Execution permissions of /Users/hazeman/Development/uniit/ygames/node_modules/.prisma/client/query-engine-darwin are fine  
  prisma:engine Client Version: 2.23.0  
  prisma:engine Engine Version: query-engine adf5e8cba3daf12d456d911d72b6e9418681b28b  
  prisma:engine Active provider: postgresql  
  prisma:engine stdout  PANIC in query-engine/core/src/interpreter/query_interpreters/nested_read.rs:79:56
1  
  prisma:engine {
  prisma:engine   error: SocketError: other side closed
  prisma:engine       at Socket.onSocketEnd (/Users/hazeman/Development/uniit/ygames/node_modules/@prisma/client/runtime/index.js:25530:24)
  prisma:engine       at Socket.emit (events.js:327:22)
  prisma:engine       at endReadableNT (internal/streams/readable.js:1327:12)
  prisma:engine       at processTicksAndRejections (internal/process/task_queues.js:80:21) {
  prisma:engine     code: 'UND_ERR_SOCKET'
  prisma:engine   }
  prisma:engine }  
```

## Client Snippet
```ts
prisma.tournamentPrize.findUnique({
  where: {
    id,
  },
  include: {
    currency: true,
    customType: true,
    winners: true,
  },
}),
```

## Relevant Part of the Schema
```prisma
model TournamentPrize {
  id            BigInt                       @id @default(autoincrement())
  createdAt     DateTime?                    @default(now()) @map(name: "created_at") @db.Timestamp(6)
  customTypeId  BigInt?                      @map(name: "custom_type_id")
  customType    TournamentCustomPrizeType?   @relation(fields: [customTypeId], references: [id])
  currencyId    BigInt?                      @map(name: "currency_id")
  currency      Currency?                    @relation(fields: [currencyId], references: [id])
  type          TournamentPrizeType
  placementType TournamentPrizePlacementType @map(name: "placement_type")
  quantity      Int
  placement     Int[]
  state         TournamentPrizeState         @default(CREATED)
  tournamentId  BigInt                       @map(name: "tournament_id")
  tournament    Tournament                   @relation(fields: [tournamentId], references: [id])
  winners       TournamentTeam[]

  @@map(name: "tournament_prizes")
}

model Tournament {
  id                      BigInt              @id @default(autoincrement())
  eventId                 BigInt?             @map(name: "event_id")
  gameName                String?             @map(name: "game_name")
  description             String?
  name                    String?
  type                    String?
  gameId                  BigInt?             @map(name: "game_id")
  capMin                  Int?                @map(name: "cap_min")
  capMax                  Int?                @map(name: "cap_max")
  date                    DateTime?           @db.Date
  startTime               String?             @map(name: "start_time")
  endTime                 String?             @map(name: "end_time")
  registrationStartTime   DateTime?           @map(name: "registration_start_time") @db.Timestamptz(6)
  registrationEndTime     DateTime?           @map(name: "registration_end_time") @db.Timestamptz(6)
  checkinStartTime        DateTime?           @map(name: "checkin_start_time") @db.Timestamptz(6)
  checkinEndTime          DateTime?           @map(name: "checkin_end_time") @db.Timestamptz(6)
  tournamentInfo          String?             @map(name: "tournament_info")
  embedStream             String?             @map(name: "embed_stream")
  partner                 String?
  partner2                String?
  specialRegistration     String?             @map(name: "special_registration")
  mailSubject             String?             @map(name: "mail_subject")
  mailText                String?             @map(name: "mail_text")
  enablePending           Boolean?            @map(name: "enable_pending")
  pendingMailSubject      String?             @map(name: "pending_mail_subject")
  pendingMailText         String?             @map(name: "pending_mail_text")
  embedTournament         Json?               @map(name: "embed_tournament")
  dateEnd                 DateTime?           @map(name: "date_end") @db.Date
  checkinStartTimePending DateTime?           @map(name: "checkin_start_time_pending") @db.Timestamptz(6)
  checkinEndTimePending   DateTime?           @map(name: "checkin_end_time_pending") @db.Timestamptz(6)
  public                  Boolean?            @default(true)
  externalTournament      String?             @map(name: "external_tournament")
  publicMatches           Boolean?            @default(false) @map(name: "public_matches")
  style                   String?
  profilePhoto            String?             @map(name: "profile_photo")
  rules                   Json?
  playersCount            Int?                @map(name: "players_count")
  contact                 String?
  adminsEmails            String[]            @map(name: "admins_emails")
  customFields            Json?               @map(name: "custom_fields")
  extrasCount             Int?                @map(name: "extras_count")
  game                    Game?               @relation(fields: [gameId], references: [revisionId])
  merchantId              BigInt?             @map(name: "merchant_id")
  merchant                Merchant?           @relation(fields: [merchantId], references: [id])
  teams                   TournamentTeam[]
  projectTournaments      ProjectTournament[]
  matches                 TournamentMatch[]
  stages                  TournamentStage[]
  prizes                  TournamentPrize[]

  @@map(name: "game_tournaments")
}

model TournamentTeam {
  tournamentId  BigInt            @map(name: "game_tournament_id")
  name          String
  players       Json?
  contact       String
  isManager     Boolean?          @map(name: "is_manager")
  createdAt     DateTime?         @default(now()) @map(name: "created_at") @db.Timestamp(6)
  schedule      String?
  extra         String?
  discord       String?
  pending       Boolean?          @default(false)
  state         Int?              @default(1)
  id            BigInt            @id @default(autoincrement())
  fixedOrder    Int?              @map(name: "fixed_order")
  ownerId       BigInt?           @map(name: "owner")
  manager       Json?
  custom        Json?
  teamId        BigInt?           @map(name: "team_id")
  tournament    Tournament        @relation(fields: [tournamentId], references: [id])
  owner         User?             @relation(fields: [ownerId], references: [id])
  team          Team?             @relation(fields: [teamId], references: [id])
  eventPayments EventPayment[]
  teamAMatches  TournamentMatch[] @relation(name: "TournamentTeamA")
  teamBMatches  TournamentMatch[] @relation(name: "TournamentTeamB")
  invitations   TeamInvitation[]
  claimedPrizes TournamentPrize[]

  @@unique([tournamentId, name], name: "game_tournament_teams_game_tournament_id_name_key")
  @@map(name: "game_tournament_teams")
}
```

**Comment by Hazeman28** [comment]
If I remove nested `winners: true` then prisma doesn't crash.

**Comment by Hazeman28** [comment]
Fixed by updating to v2.24.1
